### PR TITLE
SONAR-7571 support huge file source in FILE_SOURCES.BINARY_DATA

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/source/FileSourceDto.java [hunk 1]
```diff
@@ -36,6 +38,8 @@
 
 public class FileSourceDto {
 
+  private static final String SIZE_LIMIT_EXCEEDED_EXCEPTION_MESSAGE = "Protocol message was too large.  May be malicious.  Use CodedInputStream.setSizeLimit() to increase the size limit.";
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/source/FileSourceDto.java]: MINOR Split this 188 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: sonar-db/src/test/java/org/sonar/db/source/FileSourceDtoTest.java [hunk 2]
```diff
@@ -54,14 +60,28 @@ public void getSourceData_throws_ISE_with_id_fileUuid_and_projectUuid_in_message
...
     expectedException.expectMessage("Fail to decompress and deserialize source data [id=" + id + ",fileUuid=" + fileUuid + ",projectUuid=" + projectUuid + "]");
 
     underTest.getSourceData();
   }
+
+  @Test
+  public void getSourceData_reads_Data_object_bigger_than_default_size_limit() {
+    DbFileSources.Data.Builder dataBuilder = DbFileSources.Data.newBuilder();
+    DbFileSources.Line.Builder lineBuilder = DbFileSources.Line.newBuilder();
+    for (int i = 0; i < 199999; i++) {
```
Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/source/FileSourceDtoTest.java]: why 199999 ?

File: sonar-db/src/test/java/org/sonar/db/source/FileSourceDtoTest.java [hunk 3]
```diff
@@ -54,14 +60,28 @@ public void getSourceData_throws_ISE_with_id_fileUuid_and_projectUuid_in_message
...
+      dataBuilder.addLines(
+        lineBuilder.setSource(LOREM_IPSUM)
+          .setLine(i)
+          .build());
+    }
+    byte[] data = FileSourceDto.encodeSourceData(dataBuilder.build());
+    new FileSourceDto().decodeSourceData(data);
```
Comment (simonbrandhof) [inline sonar-db/src/test/java/org/sonar/db/source/FileSourceDtoTest.java]: no assertions ?

File: sonar-db/src/test/java/org/sonar/db/source/FileSourceDtoTest.java (hunk 2 above)
Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/source/FileSourceDtoTest.java]: because 99999 is not enough and it was easier to just add a 1 before :)

**Pull request merged**